lib/config/config.go

```diff
@@ -270,6 +270,16 @@ func (cfg *Configuration) prepare(myID protocol.DeviceID) error {
 		cfg.GUI.APIKey = rand.String(32)
 	}
 
+	// The list of ignored devices should not contain any devices that have
+	// been manually added to the config.
+	newIgnoredDevices := []protocol.DeviceID{}
```

Comment:
We want the items here to not be nil for nicer JSON serialization, which is why this is not just a `var newIgnoredDevices []protocol.DeviceID`.